engine/source/assets/ShaderAssetManager.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include "assets/AssetService.h"
+
+#include <string>
+#include <vector>
+
+namespace nc
+{    
+    class ShaderAssetManager final : public IAssetService<ShaderView, std::string>
```

Comment:
Adding to my comment on the `ShaderView`: I think this file should define a type `ShaderAssetFlyweight` which has the actual data (vectors and such). Then m_shaderViews should be a `std::vector<ShaderAssetFlyweight>` and `Acquire()` still returns a `ShaderView`, which now only has non-owning members (spans and such). Basically what `ConcaveColliderAssetManager` does.

Reply:
Also, I don't remember from a previous conversion if flyweight is the right word here, or if flyweight is synonymous with the view - I'm fine with whatever, but the current pattern is flyweight=owning +internal and view=nonowning + api facing.